docs/connectors/twitch.md

```diff
@@ -0,0 +1,78 @@
+# Twitch
+
+A connector for [Twitch](https://twitch.tv/).
+
+## Requirements
+
+- A Twitch Account
+- A Twitch App obtained from [Twitch developers page](https://dev.twitch.tv/console/apps)
+- The `code` obtained from the first step of OAuth
+
+## Configuration
+
+```yaml
+connectors:
+  twitch:
+    # required
+    code: "hfu923hfks02nd2821jfislf"  # Code obtained from the first oauth step
+    client-id: "e0asdj48jfkspod0284"
+    client-secret: "kdksd0458j93847j"
+    channel: theflyingdev # Broadcaster channel
+    redirect: http://localhost # Url to be passed to get oath token - defaults to localhost
+    foward-url: 'http://94jfsd9ff.ngrok.io' # Either an URL provided by a forwarding service or an exposed ip address
+    # optional
+    always-listening: false # Turn on to connect to the chat server even if stream is offline.
+```
+
+## Setup Twitch App
+
+You need to create a [Twitch App](https://dev.twitch.tv/console/apps) to use the Twitch Connector. Click the `+ Register Your Application` button, give this app a name and a redirect url - using `http://localhost` is fine. Once created, you can click the `Manage` button and get your `client-id`, you can then get a `client-secret` by pressing the `New Secret` button (keep this secret safe as it won't be shown to you again).
+
+## Getting OAuth code
+
+Twitch OAuth has two steps, first you need to make a `GET` request to a specific URL to obtain a `code`. After you've received the code, you need to make a `POST` request to the same URL and Twitch will send you an `access_token` and `refresh_token`. 
+
+After a certain period, the `access_token` will expire and you have to make a new request with your `refresh_token` to re-authenticate yourself. 
```

Comment:
Does the user need to do this themselves, or can the connector handle this for them?

Reply:
User's don't care about how the tokens work or how opsdroid refreshes them. They just need simple steps to get set up.